### PR TITLE
increase clang-tidy coverage in torch/csrc

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305 @ipiszy

### CODE REVIEW

**Comment by Skylion007** [comment]
The header filter reflex in .clang-tidy needs to be updated, otherwise none of these checks will run on any of the headers in the new files. Without the headers, many of the checks are implicitly disabled.

**Comment by cyyever** [comment]
Need to allocate more time to clang-tidy-checks

**Comment by albanD** [comment]
cc @malfet about lintrunner changes?

**Comment by Skylion007** [comment]
@cyyever It sounds like you might just need to add a timeout-minutes key to here: https://github.com/pytorch/pytorch/blob/f9988691600adf3f19beb1b13eaa33484e80b57e/.github/workflows/lint.yml#L20

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ on:
 jobs:
   lintrunner:
     uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    timeout-minutes: 60
```
Comment (Skylion007) [inline .github/workflows/lint.yml:19]: This key goes under the "with:" key @cyyever

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ jobs:
   lintrunner:
     uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
     with:
+      timeout-minutes: 60
```
Comment (Skylion007) [inline .github/workflows/lint.yml:20]: Huh, not sure why it's still not working. It looks like just like this snippit: https://github.com/pytorch/pytorch/blob/main/.github/workflows/slow.yml#L43

Comment (cyyever) [reply .github/workflows/lint.yml:20]: The job is not scheduled to run

Comment (cyyever) [reply .github/workflows/lint.yml]: should use 'timeout'

**Comment by Skylion007** [comment]
Maybe it would be best to start by enabling it only on a subset of torch/csrc given the number of errors / changes needed to fix it?

File: torch/csrc/Stream.cpp [hunk 3]
```diff
@@ -53,7 +54,7 @@ PyObject* THPStream_Wrap(const c10::Stream& stream) {
 
   THPStream* self = (THPStream*)ptr.get();
   self->stream_id = stream.id();
-  self->device_index = stream.device_index();
+  self->device_index = static_cast<int64_t>(stream.device_index()); // NOLINT
```
Comment (Skylion007) [inline torch/csrc/Stream.cpp:57]: NOLINT with which error code?

**Review by Skylion007 — COMMENTED** [review]
Some nits

File: torch/csrc/Stream.cpp (hunk 3 above)
Comment (cyyever) [reply torch/csrc/Stream.cpp:57]: Clang-tidy says int8_t can't be directly casted to int64_t

**Comment by Skylion007** [comment]
You have done a lot of great work, you may to consider splitting this PR into several parts to get it merged faster.

File: torch/csrc/Storage.cpp [hunk 4]
```diff
@@ -134,8 +134,8 @@ static PyObject* THPStorage_pynew(
   torch::ParsedArgs<3> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
 
-  int64_t allocator_arg_idx = 0;
-  int64_t device_arg_idx = 1;
+  int allocator_arg_idx = 0;
```
Comment (Skylion007) [inline torch/csrc/Storage.cpp:137]: Do we really want to lower it to an int?

File: c10/core/TensorImpl.h [hunk 5]
```diff
@@ -65,7 +65,7 @@ inline std::vector<int64_t> ToVectorint64_t(const ArrayRef<int>& src) {
 /**
  * Return product of all dimensions starting from k
  */
-inline int64_t size_from_dim_(int k, IntArrayRef dims) {
```
Comment (Skylion007) [inline c10/core/TensorImpl.h:68]: I think this can stay an int since we do not support int64 number of dims?

File: torch/csrc/Storage.cpp (hunk 4 above)
Comment (cyyever) [reply torch/csrc/Storage.cpp:137]: Or change interface to accept int64?

Comment (cyyever) [reply torch/csrc/Storage.cpp:137]: The following API wants int  indices. We should revise them in another PR.

File: c10/core/TensorImpl.h (hunk 5 above)
Comment (cyyever) [reply c10/core/TensorImpl.h:68]: The next function size_to_dim_ also uses int64 for k

Comment (cyyever) [reply c10/core/TensorImpl.h]: Reverted

**Review by Skylion007 — APPROVED** [review]
Nice, finally passes lintrunner, this is impressive expansion of clang-tidy coverage. We may want to invest a clang-tidy cache (ctcache) sometime in the future as this quickly becoming the heaviest lintrunner check by far.

**Comment by Skylion007** [comment]
@pytorch rebase

**Comment by cyyever** [comment]
> Nice, finally passes lintrunner, this is impressive expansion of clang-tidy coverage. We may want to invest a clang-tidy cache (ctcache) sometime in the future as this quickly becoming the heaviest lintrunner check by far.

I will add aten/src/ATen/core torch/autograd and other dirs to clang-tidy since they are so important, so it will quickly become extremely slow.

**Pull request closed without merging**